### DISCUSSION TITLE
Is there a way to throw an error if a connection is already present?

### DISCUSSION THREAD

**Discussion by nullndr** [prisma/prisma#11620] [Q&A] [answered]
When I call this function

```typescript
export const updateTagWithRole = async (tag: string, role: string) => {
  return db.tag.update({
    where: { text: tag },
    data: {
      roles: {
        connect: { text: role }
      }
    }
  })
}
```

with this schema

```prisma
model Tag {
  id        String @id @default(cuid())
  text      String @unique

  roles     Role[]
}

model Role {
  id        String @id @default(cuid())
  text      String @unique

  tags      Tag[]
}
```

The query ends even if the connection is already present, how can I throw an error?

**Comment by pantharshit00** [comment, marked as answer]
Hello @nullndr 

You will need to check the connection in another query here:
```ts
export const updateTagWithRole = async (tag: string, role: string) => {
  const existing = await prisma.tag.findMany({
    where: {
      roles: {
        some: {
          text: role,
        },
      },
    },
  });
  if (existing.length > 0) {
    throw new Error("already exists");
  }
  return prisma.tag.update({
    where: { text: tag },
    data: {
      roles: {
        connect: { text: role },
      },
    },
  });
};
```

**Reply by nullndr** [reply]
Thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
